client/state/sites/schema.js

```diff
@@ -0,0 +1,74 @@
+const icon = {
+	type: 'object',
+	properties: {
+		img: { type: 'string' },
+		ico: { type: 'string' }
+	}
+};
+
+const logo = {
```

Comment:
Personal preference here, but I think splitting with such granularity hurts readability more than it helps. If it were me, I'd limit it to `site` object at most, if not simply exporting the entire schema object:

```js
export default {
	type: 'object',
	patternProperties: {
		'^\\d+$': {
			type: 'object',
			properties: {
				ID: { type: 'number' },
				name: { type: 'string' },
				description: { type: 'string' },
				URL: { type: 'string' },
				jetpack: { type: 'boolean' },
				post_count: { type: 'number' },
				subscribers_count: { type: 'number' },
				lang: { type: 'string' },
				icon: {
					type: 'object',
					properties: {
						img: { type: 'string' },
						ico: { type: 'string' }
					}
				},
				logo: {
					type: 'object',
					properties: {
						id: { type: 'number' },
						sizes: { type: 'array' },
						url: { type: 'string' }
					}
				},
				visible: { type: 'boolean' },
				is_private: { type: 'boolean' },
				is_following: { type: 'boolean' },
				options: { type: 'object' },
				meta: { type: 'object' },
				user_can_manager: { type: 'boolean' },
				is_vip: { type: 'boolean' },
				is_multisite: { type: 'boolean' },
				capabilities: {
					type: 'object',
					patternProperties: {
						'^[a-z_]+$': { type: 'boolean' }
					}
				},
				plan: {
					type: 'object',
					properties: {
						product_id: { type: 'number' },
						product_slug: { type: 'string' },
						product_name_short: { type: 'string' },
						free_trial: { type: 'boolean' }
					}
				},
				single_user_site: { type: 'boolean' },
				domain: { type: 'string' },
				slug: { type: 'string' },
				title: { type: 'string' }
			},
			required: [ 'ID', 'name' ]
		}
	},
	additionalProperties: false
};
```